sdk/nodejs/resource.ts

```diff
@@ -750,6 +758,58 @@ export interface CustomTimeouts {
  */
 export type ResourceTransformation = (args: ResourceTransformationArgs) => ResourceTransformationResult | undefined;
 
+/**
+ * ResourceTransform is the callback signature for the `transforms` resource option.  A
+ * transform is passed the same set of inputs provided to the `Resource` constructor, and can
+ * optionally return back alternate values for the `props` and/or `opts` prior to the resource
+ * actually being created.  The effect will be as though those props and opts were passed in place
+ * of the original call to the `Resource` constructor.  If the transform returns undefined,
+ * this indicates that the resource will not be transformed.
+ */
+export type ResourceTransform = (args: ResourceTransformArgs) => Promise<ResourceTransformResult | undefined>;
```

Comment:
Made this a promise because it's running async anyway so if people want to do async work inside transforms they can easily do so now.

Reply:
But this is now one more thing we're going to have to note in the migration guide -- that you basically have to use an async function now or return a promise.

Should we make it not return Promise or make it optionally return promise?